File: Exercicios-JAVA/Filas/FilaSimples/FilaSimples.java
import java.util.Scanner;

public class FilaSimples {
    private int tam;
    private int tamTotal;
    private int[] vet;

    public FilaSimples() {
        this.tam = 0;
        this.tamTotal = 7;
        this.vet = new int[tamTotal];
    }

    public void enqueue(int valor){
        if(tam < tamTotal){
            vet[tam] = valor;
            tam++;
        }else{
            System.out.println("Fila Cheia");
        }
    }

    public void dequeue() {
        if (tam == 0) {
            System.out.println("Fila vazia!");
        } else {
            
            for (int i = 0; i < tam - 1; i++) {
                vet[i] = vet[i + 1];
            }
            tam--;
        }
    }

    public void mostrar(){
      
            for(int i = 0; i < tam; i++){
                System.out.print(vet[i] + " ");
            }
          
            System.out.println();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        FilaSimples fila = new FilaSimples();
        int opc;

        do{ 
            System.out.println("----- MENU -----");
            System.out.println("Digite 0 para SAIR");
            System.out.println("Digite 1 para INSERIR");
            System.out.println("Digite 2 para REMOVER");
            System.out.println("Digite 3 para MOSTRAR");
            System.out.print("Digite uma opcao: ");
        
            int valor;
            opc = scanner.nextInt();

            switch (opc) {
                case 1:
                    System.out.println("Digite o valor a ser inserido");
                    valor = scanner.nextInt();
                    fila.enqueue(valor);
                    break;

                case 2:
                    fila.dequeue();
                    break;
                case 3:
                    fila.mostrar();
                    break;

                default:
                    throw new AssertionError();
            }
        } while (opc != 0);
    }

    
}

File: Exercicios C/Filas/FilaSimples/FilaSimples.c
#include <stdio.h>
#include <stdlib.h>

void enqueue(int vet[], int *tam, int tamTotal,int valor){
    if(*tam < tamTotal){
            vet[*tam] = valor;
            (*tam)++;
        }else{
            printf("\nFila Cheia");
        }
}

void dequeue(int vet[], int *tam){
    if (*tam == 0) {
            printf("\nFila vazia!");
        } else {
            
            for (int i = 0; i < *tam - 1; i++) {
                vet[i] = vet[i + 1];
            }
            (*tam)--;
        }
}

void mostrar(int vet[], int tam){

    for(int i = 0; i < tam; i++){
                printf("%d ", vet[i]);
            }
          
            printf("\n");
}

int main(){
    int tam = 0;
    int tamTotal = 7;
    int vet[tamTotal];
    int opc;

    do{ 
        printf("\n----- MENU -----");
        printf("\nDigite 0 para SAIR");
        printf("\nDigite 1 para INSERIR");
        printf("\nDigite 2 para REMOVER");
        printf("\nDigite 3 para MOSTRAR");
        printf("\nDigite uma opcao: ");
        
        int valor;
        scanf("%d", &opc);

        switch (opc) {
            case 1:
                printf("\nDigite o valor a ser inserido: ");
                scanf("%d", &valor);
                enqueue(vet, &tam, tamTotal, valor);
            break;

            case 2:
                dequeue(vet, &tam);
            break;
            case 3:
                mostrar(vet,tam);
            break;

            case 0:
                printf("SAINDO....");
            break;

            default:
                printf("\nOpcao invalida!");
            }

        } while (opc != 0);
}
